### PR TITLE
Tenure investigation trigger

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/browse/MTTL-3030)
## Describe this PR

Implement Tenure Investigation trigger
#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs [hunk 1]
```diff
@@ -222,14 +224,44 @@ public async Task ProcessStateIsUpdatedToApplicationSubmittedOnSubmitApplication
...
         #endregion
 
+        #region Tenure Investigation
+
+        [Theory]
+        [InlineData(SharedValues.Approve, SharedStates.TenureInvestigationPassed)]
+        [InlineData(SharedValues.Decline, SharedStates.TenureInvestigationFailed)]
+        [InlineData(SharedValues.Appointment, SharedStates.TenureInvestigationPassedWithInt)]
+        public async Task ProcessStateIsUpdatedOnTenureInvestigationTrigger(string tenureInvestigationRecommendation, string expectedState)
+        {
```
Comment (notatiyyah) [inline ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs:242]: Do we have a test for tenure investigation recommendation invalid?

File: ProcessesApi/V1/Services/ChangeOfNameService.cs [hunk 2]
```diff
@@ -76,6 +77,25 @@ private async Task ReviewDocumentsCheck(StateMachine<string, string>.Transition
...
+
+
+            var triggerMappings = new Dictionary<string, string>
+            {
+                { SharedValues.Appointment, SharedInternalTriggers.TenureInvestigationPassedWithInt },
+                { SharedValues.Approve, SharedInternalTriggers.TenureInvestigationPassed },
+                { SharedValues.Decline, SharedInternalTriggers.TenureInvestigationFailed }
+            };
+
+            processRequest.HandleRecommendation(triggerMappings,
```
Comment (notatiyyah) [inline ProcessesApi/V1/Services/ChangeOfNameService.cs:92]: Finally thought of a good name for this method! How about `SelectTriggerFromUserInput` or select from form data?

Also does this method have unit tests?

File: ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs (hunk 1 above)
Comment (humulla) [reply ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs:242]: Yes that is added to the ThrowsFormDataNotFoundException tests

Comment (notatiyyah) [reply ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs:242]: No, I meant (like in the E2E tests), if there is a value for tenure recommendation but it's not one of approve, decline, appointment etc.

Comment (humulla) [reply ProcessesApi.Tests/V1/Services/ChangeOfNameServiceTests.cs:242]: That has been added here:
https://github.com/LBHackney-IT/processes-api/pull/113/files/b86b7810515ee0bf95b9fa70a6eb6e9aba129909#diff-ba9efb7dddcf301d44b020329213bb93bddba9fa0325f49d0bf465ab9949ff27R273

**Review by notatiyyah — APPROVED** [review]

**Pull request merged**